src/metrics/prometheus.go

```diff
@@ -205,3 +211,21 @@ func (m *metrics) pushMetrics() int {
 	log.Debug("Push #%d of metrics in %0.3fs", m.pushes, time.Since(start).Seconds())
 	return 0
 }
+
+// deriveLabelValue runs a command and returns its output.
+// It returns the empty string on error; we assume it's better to keep the set of labels constant on failure.
```

Comment:
Agreed, it's probably better to be stricter now, we can always loosen it later if it becomes a problem.